homeassistant/components/mqtt/client.py

```diff
@@ -598,15 +598,15 @@ def _mqtt_on_message(self, _mqttc, _userdata, msg) -> None:
         self.hass.add_job(self._mqtt_handle_message, msg)
 
     @lru_cache(2048)
-    def _matching_subscriptions(self, topic):
-        subscriptions = []
+    def _matching_subscriptions(self, topic: str) -> list[Subscription]:
+        subscriptions: list[Subscription] = []
         for subscription in self.subscriptions:
             if subscription.matcher(topic):
                 subscriptions.append(subscription)
         return subscriptions
 
     @callback
-    def _mqtt_handle_message(self, msg) -> None:
+    def _mqtt_handle_message(self, msg: ReceiveMessage) -> None:
```

Comment:
I think `msg` here is an instance from paho.mqtt:

https://github.com/eclipse/paho.mqtt.python/blob/9782ab81fe7ee3a05e74c7f3e1d03d5611ea4be4/src/paho/mqtt/client.py#L387-L399

https://github.com/eclipse/paho.mqtt.python#on_message